File: src/main/webapp/js/Controllers/CorController.js
/* 
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
var module = angular.module("Cor", []);

    module.service("CorService", function ($http){
        
        var url = "/cores";
        
        this.list = function(){
            return $http.get(url);
        };
        
        this.save = function(cor){
            return $http.post(url, cor);
        };
        
        this.remove = function(id){
            return $http.delete(url +'/'+ id);
        };
        
        this.getById = function(id){
            return $http.get(url + '/' + id);
        };
        
        this.edit = function(cor){
            return $http.put(url +'/'+ cor.id, cor);
        };
    });

    module.controller("CorController", function ($scope, CorService, $location, $routeParams){
        
        $scope.cores = [];
        $scope.update = false;
        
        $scope.atualizarCores = function(){
            $scope.cor = {};
            CorService.list().success(function(data){
                $scope.cores = data;
                console.log($scope.cores);
            }).error(function(){
                console.log("Não funfou!!!");
            });
        };
        
        $scope.cadastrar = function(cor){
            console.log($scope.cor);
            if (!$scope.update){
                CorService.save(cor).success(function(data){
                  $location.path("/cores");
                }).error(function(){
                    console.log("Não cadastrou!!!");
                });
            }else{
                CorService.edit(cor).success(function(data){
                   $location.path("/cores");
                   $scope.update = false;
                }).error(function(){
                    console.log("Não cadastrou!!!");
                });
            }
        }; 
        
        $scope.excluir = function(cor){
            CorService.remove(cor.id).success(function(){
                $scope.atualizarCores();
            }).error(function(){
               console.log("Não excluiu!!!!");
            });
        };
        
        if ($routeParams.id){
            var id = $routeParams.id;
            console.log(id);
            CorService.getById(id).success(function(data){                
                $scope.cor = data;                
                $scope.update = true;
            }).error(function(){
                console.log("Não funfou!!!");
            });
        }
    });

File: src/main/java/br/unicesumar/escoladeti2015base/cores/CorController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package br.unicesumar.escoladeti2015base.cores;


import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

/**
 *
 * @author Cristiano
 */
@RestController
@RequestMapping(value="/cores")
public class CorController {
    
    @Autowired
    private CorService service;
    
     @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public Cor getCor(@PathVariable String id) {
        return service.findById(id);
    }

    @RequestMapping(method = RequestMethod.GET)
    public List<Cor> getCores() {
        return service.findAll();
    }

    @RequestMapping(method = RequestMethod.POST)
    public void criarCor(@RequestBody Cor cor) {
        service.save(cor);
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
    public void excluirCor(@PathVariable String id) {
        service.remove(id);
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
    public void atualizarCor(@PathVariable String id, @RequestBody Cor cor) {
        service.remove(id);
        service.save(cor);
    }
}
